### PR TITLE
Add Draft notes resource

### PR DESCRIPTION
**Pull request by jaleszek**

### CODE REVIEW

File: app/controllers/concerns/draft_notes_mixin.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        redirect_to [namespace_name, form_answer]
+      end
+
+      format.json do
+        render json: {}
+      end
+    end
+  end
+
+  def update
```
Comment (matthewford) [inline app/controllers/concerns/draft_notes_mixin.rb]: Assignment Branch Condition size for update is too high. [16.03/15]

File: app/models/form_answer.rb [hunk 2]
```diff
@@ -46,15 +46,16 @@ class FormAnswer < ActiveRecord::Base
     has_one :promotion_eligibility, class_name: 'Eligibility::Promotion', dependent: :destroy
     has_one :audit_certificate, dependent: :destroy
     has_one :feedback, dependent: :destroy
+    has_one :draft_note, as: :notable, dependent: :destroy
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/models/form_answer.rb [hunk 3]
```diff
@@ -46,15 +46,16 @@ class FormAnswer < ActiveRecord::Base
...
 
-    has_many :form_answer_attachments
+    has_many :form_answer_attachments, dependent: :destroy
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/models/form_answer.rb [hunk 4]
```diff
@@ -46,15 +46,16 @@ class FormAnswer < ActiveRecord::Base
...
     has_many :support_letter_attachments, dependent: :destroy
 
     has_many :supporters, dependent: :destroy, autosave: true
     has_many :support_letters, dependent: :destroy
-    has_many :comments, as: :commentable
+    has_many :comments, as: :commentable, dependent: :destroy
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/models/form_answer.rb [hunk 5]
```diff
@@ -46,15 +46,16 @@ class FormAnswer < ActiveRecord::Base
...
     has_many :form_answer_transitions
-    has_many :assessor_assignments
+    has_many :assessor_assignments, dependent: :destroy
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/models/admin.rb [hunk 6]
```diff
@@ -25,4 +25,8 @@ class Admin < ActiveRecord::Base
   def lead?(*)
     true
   end
+
+  def full_name
```
Comment (RuslanHamidullin) [inline app/models/admin.rb]: Looks like it's already in UserSharedDecorator, which is included to AdminDecorator

**Comment by RuslanHamidullin** [comment]
Looks good :+1: 
Need to check specs:
```
rspec ./spec/features/assessor/fulfills_the_draft_notes_spec.rb:16 # Assessor fulfills the draft notes. fulfills the draft notes form
rspec ./spec/features/admin/form_answers/fulfills_the_draft_notes_spec.rb:16 # Admin fulfills the draft notes. fulfills the draft notes form
```

File: app/controllers/concerns/draft_notes_mixin.rb (hunk 1 above)
Comment (matthewford) [inline app/controllers/concerns/draft_notes_mixin.rb]: Assignment Branch Condition size for update is too high. [16.03/15]

File: app/models/form_answer.rb (hunk 2 above)
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/models/form_answer.rb (hunk 3 above)
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/models/form_answer.rb (hunk 4 above)
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/models/form_answer.rb (hunk 5 above)
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

**Comment by jaleszek** [comment]
@RuslanHamidullin thanks, updated, can you review ?

**Comment by RuslanHamidullin** [comment]
All good @jaleszek  :+1:

**Pull request merged**